File: backend/src/main/java/com/survey/system/service/SurveyResponseService.java
package com.survey.system.service;

import com.survey.system.model.SurveyResponse;
import com.survey.system.repository.SurveyResponseRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
@Transactional
public class SurveyResponseService {

    @Autowired
    private SurveyResponseRepository surveyResponseRepository;

    public SurveyResponse submitSurveyResponse(SurveyResponse surveyResponse) {
        return surveyResponseRepository.save(surveyResponse);
    }

    public List<SurveyResponse> getAllSurveyResponses() {
        return surveyResponseRepository.findAll();
    }

    public List<SurveyResponse> getSurveyResponsesBySurveyId(Long surveyId) {
        return surveyResponseRepository.findBySurveyId(surveyId);
    }

    public List<SurveyResponse> getSurveyResponsesByUserId(Long userId) {
        return surveyResponseRepository.findByUserId(userId);
    }

    public Optional<SurveyResponse> getSurveyResponseById(Long id) {
        return surveyResponseRepository.findById(id);
    }
}

File: frontend/src/services/surveyResponseService.js
import api from './api';

const surveyResponseService = {
  submitSurveyResponse: async (surveyResponse) => {
    const response = await api.post('/survey-responses', surveyResponse);
    return response.data;
  },

  getAllSurveyResponses: async () => {
    const response = await api.get('/survey-responses');
    return response.data;
  },

  getSurveyResponsesBySurveyId: async (surveyId) => {
    const response = await api.get(`/survey-responses/survey/${surveyId}`);
    return response.data;
  },

  getSurveyResponsesByUserId: async (userId) => {
    const response = await api.get(`/survey-responses/user/${userId}`);
    return response.data;
  },

  getSurveyResponseById: async (id) => {
    const response = await api.get(`/survey-responses/${id}`);
    return response.data;
  },
};

export default surveyResponseService;
